src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/mappers/TransitLayerMapper.java

```diff
@@ -44,40 +54,171 @@ public static TransitLayer map(Graph graph) {
         HashMap<LocalDate, List<TripPatternForDate>> tripPatternsByStopByDate;
         List<List<Transfer>> transferByStopIndex;
 
         LOG.info("Mapping transitLayer from Graph...");
 
-        stopsByIndex = mapStopsByIndex();
-        indexByStop = mapIndexByStop(stopsByIndex);
-        tripPatternsByStopByDate = mapTripPatterns(indexByStop);
-        transferByStopIndex = mapTransfers(graph.index.stopVertexForStop, stopsByIndex, indexByStop);
+        stopIndex =  new StopIndexForRaptor(graph.index.stopForId.values());
+        tripPatternsByStopByDate = mapTripPatterns(stopIndex);
+        transferByStopIndex = mapTransfers(graph.index.stopVertexForStop, stopIndex);
 
         LOG.info("Mapping complete.");
 
-        return new TransitLayer(tripPatternsByStopByDate, transferByStopIndex, stopsByIndex, indexByStop);
+        return new TransitLayer(
+                tripPatternsByStopByDate,
+                transferByStopIndex,
+                stopIndex.stops,
+                stopIndex.indexByStop
+        );
     }
 
-    private ArrayList<Stop> mapStopsByIndex() {
-        return new ArrayList<>(this.graph.index.stopForId.values());
+    /**
+     * Map pre-Raptor TripPatterns and Trips to the corresponding Raptor classes.
+     */
+    private HashMap<LocalDate, List<TripPatternForDate>> mapTripPatterns (
+            StopIndexForRaptor stopIndex) {
+        // CalendarService has one main implementation (CalendarServiceImpl) which contains a
+        // CalendarServiceData which can easily supply all of the dates. But it's impossible to
+        // actually see those dates without modifying the interfaces and inheritance. So we have
+        // to work around this abstraction and reconstruct the CalendarData.
+        // Note the "multiCalendarServiceImpl" which has docs saying it expects one single
+        // CalendarData. It seems to merge the calendar services from multiple GTFS feeds, but
+        // its only documentation says it's a hack.
+        // TODO OTP2 - This cleanup is added to the 'Final cleanup OTP2' issue #2757
+
+        // Reconstruct set of all dates where service is defined, keeping track of which services
+        // run on which days.
+        CalendarService calendarService = graph.getCalendarService();
+        Multimap<ServiceDate, FeedScopedId> serviceIdsForServiceDate = HashMultimap.create();
+
+        for (FeedScopedId serviceId : calendarService.getServiceIds()) {
+            Set<ServiceDate> serviceDatesForService = calendarService.getServiceDatesForServiceId(serviceId);
+            for (ServiceDate serviceDate : serviceDatesForService) {
+                serviceIdsForServiceDate.put(serviceDate, serviceId);
+            }
+        }
+
+        Map<org.opentripplanner.routing.edgetype.TripPattern, TripPattern> newTripPatternForOld;
+        newTripPatternForOld = mapOldTripPatternToRaptorTripPattern(
+                stopIndex,
+                graph.tripPatternForId.values()
+        );
```

Comment:
Here is an example of what I mean in the comment above. In the code above have I extracted the creation of the "new" TripPatterns into a static method - I then know that the new retuned data structure only depend on the stopIndex and the old tripPatterns (both given as arguments).

Reply:
This can certainly clarify things, but if the reader is not sure about the nature of the methods (if they might not be pure functions) then it can actually increase uncertainty, because you need to inspect the method body to ensure it is not updating or accumulating state in the TransitLayerMapper instance itself.

Once the overall logic of the Mapper is established I'm of course fine with breaking it down into smaller pieces, but I think we should inform the reader up front of each method's characteristics and (lack of) side effects.